src/TMB/exponential_integral.hpp

```diff
@@ -0,0 +1,22 @@
+#include "LocalCop/gaussian.hpp"
+
+#undef TMB_OBJECTIVE_PTR
+#define TMB_OBJECTIVE_PTR obj
+
+template<class Type>
+Type exponential_integral(objective_function<Type> *obj)
+{
+  DATA_VECTOR(lambda); // TODO: Provided code suggests it be PARAMETER_VECTOR(lambda) and etc.
```

Comment:
TMB only computes derivatives wrt `PARAMETER_*` macros.  So just replace each `DATA_VECTOR()` with `PARAMETER()`.  In particular, you only want each of the inputs to be passed as scalars.